install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -3206,6 +3206,19 @@ authentication:
             accessMode: ReadWriteOnce
             # -- StorageClass of the SPIRE server data storage
             storageClass: null
+          # -- Security context to be added to spire server pods.
+          # SecurityContext holds pod-level security attributes and common container settings.
+          # ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+          podSecurityContext: {}
+          # -- Security context to be added to spire server containers.
+          # SecurityContext holds pod-level security attributes and common container settings.
+          # This defaults to non root user and seccompProfile to `RuntimeDefault`.
+          # ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container
+          securityContext:
+            allowPrivilegeEscalation: false
+            runAsNonRoot: true
```

Comment:
having `runAsNonRoot` set to `true` results in the following error running the SPIRE server locally on kind.

```
container has runAsNonRoot and image will run as root (pod: "spire-server-0_cilium-spire(36e48818-7db4-4a4e-87a1-2ac6e5a5a27d)", container: spire-server)
```

Let's wait what the tests on CI report.

Reply:
Looks like CI is failing with the same issue

e.g. https://github.com/cilium/cilium/actions/runs/5889014601 (CI-E2E)

`cilium-spire         spire-server-0                               1/2     CreateContainerConfigError `

```
status:
...
    - image: ghcr.io/spiffe/spire-server:1.6.3@sha256:f4bc49fb0bd1d817a6c46204cc7ce943c73fb0a5496a78e0e4dc20c9a816ad7f
      imageID: ""
      lastState: {}
      name: spire-server
      ready: false
      restartCount: 0
      started: false
      state:
        waiting:
          message: 'container has runAsNonRoot and image will run as root (pod: "spire-server-0_cilium-spire(6938b1d5-0598-42ab-a23e-5af71f62f7fc)",
            container: spire-server)'
          reason: CreateContainerConfigError
```

Reply:
I'd recommend to either changing the default to the recommended approach in the [related issue](https://github.com/cilium/cilium/issues/27130) (i haven't tested this yet locally) or leaving it empty (and therefore up to the user configuring it to its needs)

Reply:
Done, remove the defaults. now it's flexible for the user.

Reply:
thanks!